packages/modular-scripts/src/__tests__/options.test.ts

```diff
@@ -0,0 +1,57 @@
+import os from 'os';
+import {
+  computeConcurrencyOption,
+  validateCompareOptions,
+} from '../utils/options';
+
+// Mocking process.exit and process.stderr.write for testing
+const exitSpy = jest.spyOn(process, 'exit').mockImplementation(() => {
+  throw new Error('Exit called');
+});
+const stderrSpy = jest.spyOn(process.stderr, 'write').mockImplementation();
+
+describe('validateCompareOptions', () => {
+  afterEach(() => {
+    exitSpy.mockClear();
+    stderrSpy.mockClear();
+  });
+
+  it('should exit process when compareBranch is defined and changed is false', () => {
+    expect(() => validateCompareOptions('branch', false)).toThrow(
+      'Exit called',
+    );
+    expect(stderrSpy).toHaveBeenCalledWith(
+      "Option --compareBranch doesn't make sense without option --changed\n",
+    );
+  });
+
+  it('should not exit process when compareBranch is undefined', () => {
+    expect(() => validateCompareOptions(undefined, false)).not.toThrow();
+    expect(exitSpy).not.toHaveBeenCalled();
+  });
+});
+
+describe('computeConcurrencyOption', () => {
+  afterEach(() => {
+    exitSpy.mockClear();
+    stderrSpy.mockClear();
+  });
+
+  it('should return the number of CPUs when concurrencyLevel is undefined', () => {
+    const cpus = os.cpus().length;
```

Comment:
As this test is so simple, it's hard to avoid reimplementing the function. I think you have 2 options:

1. Include `os.cpus().length || 1` in this variable so as to mimic the intended implementation as closely as possible (breaking this up increases the chance of bugs creeping in).
2. Mock `os.cpus()` to return different values, e.g. test for `2` and sometimes `undefined`. This allows us to test for more possibilities.